### ISSUE TITLE
Arch Linux OpenStack started failing at `systemctl enable cloud-init`

### ISSUE THREAD

**Issue by hartwork** [hartwork/image-bootstrap#515]
```console
[..]
# chroot /mnt/tmpc1r4v439 systemctl enable cloud-init-local
Created symlink '/etc/systemd/system/cloud-init.target.wants/cloud-init-local.service' -> '/usr/lib/systemd/system/cloud-init-local.service'.
Making service "cloud-init" start automatically...
# chroot /mnt/tmpc1r4v439 systemctl enable cloud-init
Failed to enable unit: Unit cloud-init.service does not exist
[..]
```

Related:
- https://wiki.archlinux.org/title/Cloud-init#Systemd_integration

If anyone gets a chance to debug this before I do, that would be great :pray: 

PS: For recent related CI error logs:
- https://github.com/hartwork/image-bootstrap/actions/runs/11333938928/job/31519085897#step:7:3819

**Comment by ctx** [comment]
It seems that `cloud-init-24.3.1-1` is not working well under Archlinux: https://gitlab.archlinux.org/archlinux/packaging/packages/cloud-init/-/issues/3

But it seems that before we can hit this bug we have to adjust the service names according to the release notes here: https://github.com/canonical/cloud-init/commit/143bc9e40f7f33695216db60f73e10a900d1d1cd

I think enabling `cloud-init-main.service` would be enough.

**Comment by hartwork** [comment]
@ctx excellent research, very helpful — thank you! :pray: I went for the minimal fix to just follow the rename in #517.

**Comment by vdloo** [comment]
@hartwork I noticed that because you only enabled `cloud-init-network` instead of `cloud-init-main` in https://github.com/hartwork/image-bootstrap/pull/517/files to replace the previously enabled (and no longer existing) `cloud-init` service, the behavior changed where previously with the generated images it was possible to use a `cidata` labeled `/dev/sr0` device for NoCloud cloud-init. I'm using `terraform-provider-libvirt` and cloudinit with an archlinux image stopped working a while back and it looks like this seems to be the reason.

I'm using this functionality as described in their documentation:
https://github.com/dmacvicar/terraform-provider-libvirt/blob/11dbcbf9e35ff0a7cd7594ad6382d67f69f59d7f/website/docs/r/cloudinit.html.markdown?plain=1#L6
https://github.com/dmacvicar/terraform-provider-libvirt/blob/master/examples/v0.12/ubuntu/ubuntu-example.tf

So to summarize:

before:
- an ISO with user-data could be attached as /dev/sr0 and that would be used by cloud-init on boot

after:
- cloud-init in the archlinux image no longer picks up that config, unless first `systemctl enable cloud-init-main` is done in the image

Just wanted to report this here in case this was not intended behavior

**Comment by hartwork** [comment]
Hi @vdloo, thanks for bringing this up. Please create a dedicated issue for this — a closed issue is not a good place for anything but maybe a reference to new related issues, as far as I'm concerned. PS: What you describe sounds like a [config drive](https://github.com/canonical/cloud-init/blob/e2bef78153b7621d3c0da8a2cea46bf2036ff8ae/cloudinit/sources/DataSourceConfigDrive.py#L27-L30) to me and I'd like to understand what term "cidata" means and where I can read more about it — Google didn't help me much about it. If you could add that to the new issue, that would be great. Thanks! 🙏

**Comment by vdloo** [comment]
Hi @hartwork, information about the `cidata` filesystem volume label thing can be found [here](https://cloudinit.readthedocs.io/en/latest/reference/datasources/nocloud.html#source-2-drive-with-labeled-filesystem) and [here](https://cloudinit.readthedocs.io/en/17.2/topics/datasources/nocloud.html).

**Comment by hartwork** [comment]
@vdloo so `NoCloud` and not a config drive, I see. Thanks 👍
